pkg/registry/generic/registry/storage_factory.go

```diff
@@ -30,7 +30,8 @@ func StorageWithCacher(
 	objectType runtime.Object,
 	resourcePrefix string,
 	scopeStrategy rest.NamespaceScopedStrategy,
-	newListFunc func() runtime.Object) storage.Interface {
+	newListFunc func() runtime.Object,
+	indices ...storage.IndexingFunc) storage.Interface {
```

Comment:
Deleting previous comment. I think you should have one indexfunc per collection, and it should return a list (or map, but list is probably more memory efficient) of index/value pairs for an object.